runtime/v2/shim/util.go

```diff
@@ -121,16 +122,27 @@ func WritePidFile(path string, pid int) error {
 	if err != nil {
 		return err
 	}
-	tempPath := filepath.Join(filepath.Dir(path), fmt.Sprintf(".%s", filepath.Base(path)))
+	tempPath := filepath.Join(filepath.Dir(path), "."+filepath.Base(path))
 	f, err := os.OpenFile(tempPath, os.O_RDWR|os.O_CREATE|os.O_EXCL|os.O_SYNC, 0666)
 	if err != nil {
 		return err
 	}
-	_, err = fmt.Fprintf(f, "%d", pid)
-	f.Close()
-	if err != nil {
+	needClose := true
+	defer func() {
+		if needClose {
+			f.Close()
+		}
+	}()
+
+	if _, err = f.WriteString(strconv.Itoa(pid)); err != nil {
 		return err
```

Comment:
Why not just have an `f.Close()` before this line, adding a defer and extra variable increases code complexity just to handle a single condition. I don't think there is ever a need to add more logic in between the write and close that would necessitate using the defer pattern.

Reply:
+1

Reply:
+1. Since we need to call `f.Close()` anyway, why don't we keep it the same as before:

```
_, err = f.WriteString(strconv.Itoa(pid))
closeErr := f.Close()
if err != nil {
  return err
}
if closeErr != nil {
  return closeErr
}
```

`closeErr` is only added to align with the behavior of this PR. We don't even need it if we want to keep the behavior the same as the original code (i.e. ignoring the error from `f.Close()`).

Reply:
got it
(Sorry, there are some other things being dealt with recently.)